jwst/source_catalog/source_catalog_step.py

```diff
@@ -119,4 +119,9 @@ def process(self, input_model):
                 model.meta.source_catalog = os.path.basename(cat_filepath)
                 self.log.info(f'Wrote source catalog: {cat_filepath}')
 
+                segm_model = datamodels.ImageModel(segment_img.data)
+                segm_model.update(model, only="PRIMARY")
+                self.save_model(segm_model, suffix='segm')
+                self.log.info('Wrote segmentation map to segm.fits.')
```

Comment:
Include the segment file name in the log message. After saving, it should be available in `segm_model.filename`.  May be `segm_model.meta.filename`.